src/test/java/org/candlepin/resource/GuestIdResourceTest.java

```diff
@@ -147,15 +151,13 @@ public void setUp() {
 
     @Test
     public void getGuestIdsEmpty() {
-        CandlepinQuery<GuestId> query = mock(CandlepinQuery.class);
-        CandlepinQuery<GuestIdDTO> dtoQuery = mock(CandlepinQuery.class);
-        when(guestIdCurator.listByConsumer(eq(consumer))).thenReturn(query);
-        when(query.transform((any(ElementTransformer.class)))).thenReturn(dtoQuery);
-
-        CandlepinQuery<GuestIdDTOArrayElement> result = resource.getGuestIds(consumer.getUuid());
+        List<GuestId> queryList = spy(new LinkedList<>());
+        List<GuestIdDTOArrayElement> dtoQueryList = new LinkedList<>();
+        when(guestIdCurator.listByConsumer(eq(consumer))).thenReturn(queryList);
 
-        verify(query, times(1)).transform(any(ElementTransformer.class));
-        assertEquals(result, dtoQuery);
+        Stream<GuestIdDTOArrayElement> result = resource.getGuestIds(consumer.getUuid());
+        verify(queryList, times(1)).stream();
```

Comment:
We do not care if the `.stream` method was called. This check locks us into a specific implementation for processing the list, rather than just verifying the output is correct.